client/src/pages/HomePage.test.js

```diff
@@ -0,0 +1,170 @@
+                    data: {
+                        success: true,
+                        category:
+                            [
+                                { _id: '1', name: 'Category 1' },
+                                { _id: '2', name: 'Category 2' },
+                            ]
+                    }
+                });
+            }
+            if (url === "/api/v1/product/product-list/1") {
+                return Promise.resolve({
+                    data: {
+                        success: true,
+                        products:
+                            [
+                                {
+                                    _id: '1',
+                                    name: 'Product 1',
+                                    price: 20,
+                                    description: 'Description 1',
+                                    category: '1'
+                                },
+                                {
+                                    _id: '2',
+                                    name: 'Product 2',
+                                    price: 100,
+                                    description: 'Description 2',
+                                    category: '2'
+                                }
+                            ]
+                    }
+                });
+            }
+            if (url === "/api/v1/product/product-count") {
+                return Promise.resolve({
+                    data: { total: 2 }
+                })
+            }
+        });
+        axios.post.mockResolvedValueOnce({
+            data: {
+                products: [
+                    {   
+                        _id: '1',
+                        name: 'Product 1',
+                        price: 20,
+                        description: 'Description 1',
+                        category: '1' 
+                    },
+                ],
+            },
+        });
+    });
+
+    test('should filter products based on selected categories', async () => {
+        renderComponent();
+
+        await screen.findByText('Category 1');
+        await screen.findByText('Category 2');
+
+        fireEvent.click(screen.getByText('Category 1'));
+
+        expect(await screen.findByText('Product 1')).toBeInTheDocument();
+        await waitFor(() => {
+            expect(screen.queryByText('Product 2')).not.toBeInTheDocument();
+        });
+    })
+
+    test('should filter products based on prices', async () => {
+        renderComponent();
+
+        await screen.findByText('$0 - $50');
+        await screen.findByText('$51 - $110');
+
+        fireEvent.click(screen.getByText('$0 - $50'));
+
+        expect(await screen.findByText('Product 1')).toBeInTheDocument();
+        await waitFor(() => {
+            expect(screen.queryByText('Product 2')).not.toBeInTheDocument();
+        });
+    })
+
+    test.failing('should go to next page when click loadmore button', async () => {
```

Comment:
May I ask why did this test not pass?

Reply:
The last check checks that the load more button should not be in the document as there are no more items to display but the load more button still exists.